Source/utils/string_or_view.hpp

```diff
@@ -39,9 +40,23 @@ class StringOrView {
 
 	StringOrView &operator=(StringOrView &&other) noexcept
 	{
+		string_view::size_type end = 0;
+		for (unsigned i = 0; i < 5; ++i) {
+			if (end < other.str().size()) {
+				if (other.str().find('\n', end + 1) == std::string::npos)
+					end = other.str().size();
+				else
+					end = other.str().find('\n', end + 1);
+			} else
+				break;
+		}
+		if (end < other.str().size()) {
+			LogWarn("PrintInfo unable to render everything - not enough lines");
+		}
+
 		if (owned_) {
 			if (other.owned_) {
-				str_ = std::move(other.str_);
+				str_ = std::string(other.str().substr(0, end));

```

Comment:
This eliminates the benefit of the move constructor, please revert the changes in this file.